src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -46,6 +55,41 @@ def get(self, request, organization):
             status__in=[RuleStatus.ACTIVE, RuleStatus.INACTIVE], project__in=project_ids
         )
 
+        teams = request.GET.getlist("team", [])
+        if teams:
+            unassigned = None
+            if "unassigned" in teams:
+                teams.remove("unassigned")
+                unassigned = Q(owner_id=None)
+
+            if "myteams" in teams:
+                teams.remove("myteams")
+                for t in Team.objects.filter(
+                    id__in=OrganizationMemberTeam.objects.filter(
+                        organizationmember__in=OrganizationMember.objects.filter(
+                            user=request.user, organization_id=organization.id
+                        ),
+                        is_active=True,
+                    ).values("team")
+                ):
+                    teams.append(t.id)
+
+            for team_id in teams:
+                if type(team_id) is not int and not team_id.isdigit():
+                    return Response(
+                        f"Invalid Team ID: {team_id}", status=status.HTTP_400_BAD_REQUEST
+                    )
```

Comment:
We need to validate that the user has access to these teams.

Reply:
Is this accomplished properly with `team.has_access(request.user)`?

Reply:
Also moved this logic around. In the case the user doesn't have access to one of the requested teams, it will get short circuited sooner and we'll same some Rule/AlertRule queries